src/store/account/utils.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { filter, findIndex, map, reduce } from 'lodash';
+
+import type { EditSettingsBatchResponse } from '../../network/edit-settings';
+import type {
+	AccountSettingsAttrs,
+	AccountSettingsPrefs,
+	AccountState,
+	Identity,
+	ZimletProp
+} from '../../types/account';
+import type { Mods } from '../../types/network';
+
+export function mergePrefs(mods: Partial<Mods>, state: AccountState): AccountSettingsPrefs {
+	return reduce(
+		mods.prefs,
+		(acc, pref, key) => ({
+			...acc,
+			[key]: pref as string
+		}),
+		state.settings.prefs
+	);
+}
+
+export function mergeProps(mods: Partial<Mods>, state: AccountState): Array<ZimletProp> {
+	return reduce(
+		mods.props,
+		(acc, { app, value }, key) => {
+			const newPropValue = {
+				name: key,
+				zimlet: app,
+				_content: value as string
+			};
+			const propIndex = findIndex(acc, (p) => p.name === key && p.zimlet === app);
+			if (propIndex >= 0) {
+				return acc.map((prop, index) => (propIndex === index ? newPropValue : prop));
+			}
+			return [...acc, newPropValue];
+		},
+		state.settings.props
+	);
+}
+
+export function mergeAttrs(mods: Partial<Mods>, state: AccountState): AccountSettingsAttrs {
+	return reduce(
+		mods.attrs,
+		(acc, attr, key) => ({
+			...acc,
+			[key]: attr as string
```

Comment:
the cast is not right, either the account setting attr should become unknown, or the mods one should become string.
We could probably decouple the mods arg from the Mods type, and pass only the attrs as argument, giving the function argument a smaller type.

In general, since the types are so mixed inside shell, I prefer having a bit of repetition in favor of the definition of better smaller types. I think that by doing so we can start isolating stuff.

Reply:
done